dart/lib/src/scope.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:collection';
+
+import 'protocol.dart';
+import 'sentry_options.dart';
+
+/// Scope data to be sent with the event
+class Scope {
+  /// How important this event is.
+  SeverityLevel level;
+
+  /// The name of the transaction which generated this event,
+  /// for example, the route name: `"/users/<username>/"`.
+  String transaction;
+
+  /// Information about the current user.
+  User user;
+
+  /// Used to deduplicate events by grouping ones with the same fingerprint
+  /// together.
+  ///
+  /// Example:
+  ///
+  ///     // A completely custom fingerprint:
+  ///     var custom = ['foo', 'bar', 'baz'];
+  List<String> fingerprint;
+
+  /// List of breadcrumbs for this scope.
+  ///
+  /// See also:
+  /// * https://docs.sentry.io/enriching-error-data/breadcrumbs/?platform=javascript
+  final Queue<Breadcrumb> _breadcrumbs = Queue();
+
+  /// Unmodifiable List of breadcrumbs
+  List<Breadcrumb> get breadcrumbs => List.unmodifiable(_breadcrumbs);
+
+  /// Name/value pairs that events can be searched by.
+  final Map<String, String> tags = {};
+
+  /// Arbitrary name/value pairs attached to the scope.
+  ///
+  /// Sentry.io docs do not talk about restrictions on the values, other than
+  /// they must be JSON-serializable.
+  final Map<String, dynamic> extra = {};
+
+  // TODO: eventProcessors, Contexts, BeforeBreadcrumbCallback, Breadcrumb hint, clome
+
+  final SentryOptions _options;
+
+  Scope(this._options) : assert(_options != null, 'SentryOptions is required');
```

Comment:
@rxlabz whats the state of art here for required fields? `@required` or `assert` and boom?
afaik `@required` is only used for named params

Reply:
TIL, thanks for the info :)